pkg/pillar/cmd/volumemgr/handlevolume.go

```diff
@@ -195,13 +195,13 @@ func lookupVolumeConfig(ctx *volumemgrContext,
 	return &config
 }
 
-func deleteVolume(ctx *volumemgrContext, status *types.VolumeStatus) {
+func maybeDeleteVolume(ctx *volumemgrContext, status *types.VolumeStatus) {
 
-	log.Infof("deleteVolume for %v", status.Key())
+	log.Infof("maybeDeleteVolume for %v", status.Key())
 	if status.RefCount != 0 {
 		publishVolumeStatus(ctx, status)
-		updateVolumeRefStatus(ctx, status)
-		log.Infof("deleteVolume for %v Done", status.Key())
+		updateVolumeStatusRefCount(ctx, status) // XXX needed?
```

Comment:
```maybeDeleteVolume``` is called after updating the volume status ref count so no need to update here. We need to update the volume ref status from here.

Reply:
ok

Reply:
Fixed